src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -513,6 +524,74 @@ def getThresholds(self):
 
         return
 
+    def _getJobSubmitCondition(self, jobPrio, siteName, jobType):
+        """
+        returns the string describing whether a job is ready to be submitted or the reason can't be submitted
+        Only jobs with "JobSubmitReady" return value will be added to submit job.
+        Other return values will indicate the reason jobs cannot be submitted.
+        i.e. "NoPendingSlot"  - pending slot is full with pending job
+        """
+        try:
+            totalPendingSlots = self.currentRcThresholds[siteName]["total_pending_slots"]
+            totalPendingJobs = self.currentRcThresholds[siteName]["total_pending_jobs"]
+            totalRunningSlots = self.currentRcThresholds[siteName]["total_running_slots"]
+            totalRunningJobs = self.currentRcThresholds[siteName]["total_running_jobs"]
+            highestPriorityInJobs = self.currentRcThresholds[siteName]['wf_highest_priority']
+
+            taskPendingSlots = self.currentRcThresholds[siteName]['thresholds'][jobType]["pending_slots"]
+            taskPendingJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_pending_jobs"]
+            taskRunningSlots = self.currentRcThresholds[siteName]['thresholds'][jobType]["max_slots"]
+            taskRunningJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_running_jobs"]
+
+            totalPendingJobs = self.currentRcThresholds[siteName]["total_pending_jobs"]
+            # set the initial totalPendingJobs since it increases in every cycle when a job is submitted
+            self.currentRcThresholds[siteName].setdefault("init_total_pending_jobs", totalPendingJobs)
+
+            taskPendingJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_pending_jobs"]
+            # set the initial taskPendingJobs since it increases in every cycle when a job is submitted
+            self.currentRcThresholds[siteName]['thresholds'][jobType].setdefault("init_task_pending_jobs", taskPendingJobs)
+
+            initTotalPending = self.currentRcThresholds[siteName]["init_total_pending_jobs"]
+            initTaskPending = self.currentRcThresholds[siteName]['thresholds'][jobType]["init_task_pending_jobs"]
+
+        except KeyError as ex:
+            msg = "Invalid key for site %s and job type %s\n" % (siteName, jobType)
+            msg += traceback.format_exc()
```

Comment:
not sure if you forgot to replace it by logging.exception or just didn't want to. Ping'ing you again just to be sure.

Reply:
Of course I forgot. :-)